### PR TITLE
Update to 5.12

### PR DESCRIPTION
**Pull request by ccordoba12**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

 
Closes #60.

### CODE REVIEW

**Comment by ccordoba12** [comment]
@conda-forge-admin, please rerender

**Comment by ccordoba12** [comment]
@isuruf, @ocefpaf, there are a couple of issues to consider here:

* PyQt 5.12 split QtWebEngine into a different package (pyqtwebengine) to reduce size installation. That's good for packages like Matplotlib or qtconsole, which don't require to render web content in Qt widgets. Others (like Spyder) require that extra functionality.

    However, I don't know how to handle conditional dependencies in conda to support previous Qt versions. That's because if PyQt < 5.12, then there's no need for packages like Spyder to require `pyqtwebengine`. But if PyQt >= 5.12, then we need to require that package or Spyder will fail to start.

    What should we do? Should we create a different feedstock for `pyqtwebengine` or not?

* There's a strange bug on Windows when running `configure.py` (I think), which seems to point out to a faulty Qt build:

    > Error: Failed to determine the detail of your Qt installation. Try again using
the --verbose flag to see more detail about the problem.

**Comment by ccordoba12** [comment]
> There's a strange bug on Windows when running configure.py

Sorry, forget that comment. That's only seen in Azure but not in Appveyor (don't know why though).

**Comment by isuruf** [comment]
For 1, you can create an empty `pyqtwebengine` for `qt<5.12` and require `pyqtwebengine` in `spyder`

File: recipe/meta.yaml [hunk 1]
```diff
@@ -53,13 +53,14 @@ requirements:
     - pkg-config  # [not win]
   host:
     - python
-    - sip >=4.19.4,<=4.19.8
+    - enum34  # [py27]
+    - sip 4.19.18
     - dbus  # [not win]
-    - qt 5.9.*
+    - qt 5.12
```
Comment (jschueller) [inline recipe/meta.yaml]: it should be `5.12.*`, else conda emits a warning

**Comment by ccordoba12** [comment]
> For 1, you can create an empty pyqtwebengine for qt<5.12

For that we would have to split QtWebEngine from PyQt 5.9 too, right? But I don't know how to do that.

File: recipe/meta.yaml (hunk 1 above)
Comment (ccordoba12) [reply recipe/meta.yaml]: Ok, I thought it was the other way around. Will fix it.

**Comment by isuruf** [comment]
> For that we would have to split QtWebEngine from PyQt 5.9 too, right? But I don't know how to do that.

No. For Qt 5.9, `pyqtwebengine=5.9` will require `pyqt=5.9` and that's it. It'll be an empty package with just the requirements.

**Comment by ccordoba12** [comment]
Ok, got it. Thanks for the advice @isuruf!

Unfortunately, I don't have time to submit `pyqtwebengine` to staged-recipes. Besides, I don't know how to compile it, i.e. if it requires PyQt5 sources or not.

Could someone help me with that?

**Comment by isuruf** [comment]
`pyqtwebengine` manylinux wheel without the qtwebengine libraries is only half a MB, so we might as well have `pyqtwebengine` in `pyqt` until we split `qt`

**Comment by ccordoba12** [comment]
Actually pyqtwebengine weights ~50 Mb:

Selección_009

**Comment by isuruf** [comment]
If you download it, you'll see that most of that is a folder called `PyQt5/Qt` which is in our `qt=5.12` package. The 3 other files in `PyQt5` are less than half a MB.

image

**Comment by ccordoba12** [comment]
> you'll see that most of that is a folder called PyQt5/Qt which is in our qt=5.12 package

Yeah, you're right. Those libraries are part of our Qt package, so there's no need to split this one.

I'll continue with this PR then when I have more free time next week.

**Comment by pkgw** [comment]
@ccordoba12 Is there anything people can do to help with this?

**Comment by ccordoba12** [comment]
You could open a new PR for this, if you want, because I'm really swamped right now, sorry.

**Comment by leofang** [comment]
@ccordoba12 could you reopen #61 then?

**Comment by ccordoba12** [comment]
@leofang, that was the wrong approach. You need to take what I've done here and try to finish it.

**Comment by leofang** [comment]
No, as I said there you need a private copy of sip. Your approach cannot achieve that. I can try to combine both approaches to support the WebEngine, but a special handling of sip is a must.

**Comment by ccordoba12** [comment]
> No, as I said there you need a private copy of sip

Why not an installed version of sip? And if a private copy is inevitable, is it not build as part of PyQt5?

**Comment by leofang** [comment]
> > No, as I said there you need a private copy of sip
> 
> Why not an installed version of sip? And if a private copy is inevitable, is it not build as part of PyQt5?

See conda-forge/sip-feedstock#16 for the references. PyQt changed the build requirements in 5.10(?). I didn’t invent it.

**Comment by isuruf** [comment]
@leofang, does a symlink from `site-packages/sip` to `site-package/Qt5/sip` work?

**Comment by isuruf** [comment]
Or a simple python file `PyQt5/sip.py` that imports all the functions from `sip`?

**Comment by leofang** [comment]
I doubt a simple `from sip import *` statement in `PyQt5/sip.py` would work. Take a look at the file `configure.py` from sip v4.19.18, the build option `--sip-module` is converted to a macro `SIP_MODULE_NAME` in the generated makefile. Not sure how it'd be used during build time of sip and PyQt5. Also, files like `.dist-info` and `sip.pyi` also depend on the `--sip-module` option and thus need to be taken care of. 

Symlinking all relevant files might work, but I don't know enough how to overwrite paths in a conda build script. I do not have enough time to explore along this path. 

One thing I'd like to note: I admit the `wget` command in #61 isn't legit. On some systems it'd fail. But as I marked there, it was a WIP.

**Comment by hhslepicka** [comment]
@leofang if the wget is the issue, you could add `wget` as a build dependency and run a python command to just download the proper sip package. https://anaconda.org/conda-forge/wget

**Comment by leofang** [comment]
Based on the above discussion on `pyqtwebengine` it seems the proposed workaround is to extract `PyQt5/QtWebEngine*.so` from the wheel, put them into the built PyQt5, and neglect the rest of the files (`PyQt5/Qt/*`). 

However, this doesn't work, as the three `.so` shared libraries are linked to `PyQt5/Qt/lib/libQt5*.so` (checked using `ldd`). It seems to me one just has to pip-install the `pyqtwebengine` wheel, unless the linking can be redirected to conda-forge's Qt.

**Comment by leofang** [comment]
> @leofang if the wget is the issue, you could add `wget` as a build dependency and run a python command to just download the proper sip package. https://anaconda.org/conda-forge/wget

Thanks for the suggestion, but I think this route is difficult to manage. I found that specifying multiple sources in the recipe is much simpler and robust.

**Comment by ccordoba12** [comment]
> Based on the above discussion on pyqtwebengine it seems the proposed workaround is to extract PyQt5/QtWebEngine*.so from the wheel

No, that's not what needs to be done. You need to extract pyqtwebengine sources in this recipe so that module is compiled and packaged as part of this package.

**Comment by leofang** [comment]
pyqtwebengine source is unavailable either on PyPi or on the official website if I'm not mistaken. Isn't this the reason you guys were looking at the wheel?

**Comment by ccordoba12** [comment]
> on the official website if I'm not mistaken

It's on the PyQt website, you need to look for it.

**Comment by leofang** [comment]
OK found it. I looked at the wrong page. Life is a bit easier with source code available. Thanks.

**Pull request closed without merging**